sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -137,30 +137,30 @@ object PartialAggregation {
       // Only do partial aggregation if supported by all aggregate expressions.
       if (allAggregates.size == partialAggregates.size) {
         // Create a map of expressions to their partial evaluations for all aggregate expressions.
-        val partialEvaluations: Map[TreeNodeRef, SplitEvaluation] =
-          partialAggregates.map(a => (new TreeNodeRef(a), a.asPartial)).toMap
+        val partialEvaluations: Map[ExpressionEquals, SplitEvaluation] =
```

Comment:
Why replace `TreeNodeRef` here? Even two `AttributeReference`s have exactly same name, `TreeNodeRef` will regard them as different. You are changing this behavior.